src/PowerShellEditorServices.Hosting/EditorServicesLoader.cs

```diff
@@ -0,0 +1,361 @@
+        /// <param name="hostConfig">The host configuration to start editor services with.</param>
+        /// <param name="sessionFileWriter">The session file writer to write the session file with.</param>
+        /// <returns></returns>
+        public static EditorServicesLoader Create(
+            HostLogger logger,
+            EditorServicesConfig hostConfig,
+            ISessionFileWriter sessionFileWriter,
+            IReadOnlyCollection<IDisposable> loggersToUnsubscribe)
+        {
+            if (logger == null)
+            {
+                throw new ArgumentNullException(nameof(logger));
+            }
+
+            if (hostConfig == null)
+            {
+                throw new ArgumentNullException(nameof(hostConfig));
+            }
+
+#if CoreCLR
+            // In .NET Core, we add an event here to redirect dependency loading to the new AssemblyLoadContext we load PSES' dependencies into
+
+            logger.Log(PsesLogLevel.Verbose, "Adding AssemblyResolve event handler for new AssemblyLoadContext dependency loading");
+
+            var psesLoadContext = new PsesLoadContext(s_psesDependencyDirPath);
+
+            AppDomain.CurrentDomain.AssemblyLoad += (object sender, AssemblyLoadEventArgs args) =>
+            {
+                logger.Log(
+                    PsesLogLevel.Diagnostic,
+                    $"Loaded into load context {AssemblyLoadContext.GetLoadContext(args.LoadedAssembly)}: {args.LoadedAssembly}");
+            };
+
+            AssemblyLoadContext.Default.Resolving += (AssemblyLoadContext defaultLoadContext, AssemblyName asmName) =>
+            {
+                logger.Log(PsesLogLevel.Diagnostic, $"Assembly resolve event fired for {asmName}");
+
+                // We only want the Editor Services DLL; the new ALC will lazily load its dependencies automatically
+                if (!string.Equals(asmName.Name, "Microsoft.PowerShell.EditorServices", StringComparison.Ordinal))
+                {
+                    return null;
+                }
+
+                string asmPath = Path.Combine(s_psesDependencyDirPath, $"{asmName.Name}.dll");
+
+                logger.Log(PsesLogLevel.Verbose, "Loading PSES DLL using new assembly load context");
+
+                return psesLoadContext.LoadFromAssemblyPath(asmPath);
+            };
+#else
+            // In .NET Framework we add an event here to redirect dependency loading in the current AppDomain for PSES' dependencies
+            logger.Log(PsesLogLevel.Verbose, "Adding AssemblyResolve event handler for dependency loading");
+
+            AppDomain.CurrentDomain.AssemblyLoad += (object sender, AssemblyLoadEventArgs args) =>
+            {
+                logger.Log(
+                    PsesLogLevel.Diagnostic,
+                    $"Loaded {args.LoadedAssembly.GetName()}");
+            };
+
+            AppDomain.CurrentDomain.AssemblyResolve += (object sender, ResolveEventArgs args) =>
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the type specification; it is redundant.](https://app.codacy.com/app/TylerLeonhardt/PowerShellEditorServices/pullRequest?prid=4579423)